src/resources.rs

```diff
@@ -284,6 +291,54 @@ pub fn add_repl(repl: Repl) -> Resource {
   Resource { rid }
 }
 
+pub fn add_worker(wc: WorkerChannels) -> Resource {
+  let rid = new_rid();
+  let mut tg = RESOURCE_TABLE.lock().unwrap();
+  let r = tg.insert(rid, Repr::Worker(wc));
```

Comment:
Same question as we had once with HTTP -- where does the worker get removed from the resource table?

Reply:
They weren't being correctly removed. I've fixed this now. When the thread exits it calls `resource.close()` which removes it from the table. I've added a test to ensure this.